File: client/src/app/services/CarRental/car-rental.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { CarRental } from '../../models/carRentals';

@Injectable({
  providedIn: 'root'
})
export class CarRentalService {
  constructor(private http:HttpClient) {}
  private url="http://localhost:5253/api/carRentals";
  getFromQuery(query:any):Observable<any>
  {
    return this.http.get(`${this.url}/getFromQuery?AvailableFromDate=${query.pickupdate}&AvailableUntilDate=${query.dropoffdate}&AvailableFromTime=${query.pickuptime}&AvailableUntilTime=${query.dropofftime}&location=${query.location}`);
  }
  getLocations():Observable<any>
  {
    return this.http.get(`${this.url}/getLocations`);
  }
  getAllCarRentals(pageNumber:number,pageSize:number):Observable<any>
  {
    
    return this.http.get(`${this.url}/getAllCarRentals?pageNumber=${pageNumber.toString()}&pageSize=${pageSize.toString()}`);
  }
  getById(id:any):Observable<any>
  {
    
    return this.http.get(`${this.url}/getById?id=${id}`);
  }
  createCarRental(carRentalData:CarRental)
  {
    return this.http.post(`${this.url}/createCarRental`,carRentalData);
  }
  searchByCarName(name:any):Observable<any>
  {
    return this.http.get(`${this.url}/searchByCarName?name=${name}`);
  }
updateCarRental(id:any,carRentalData:CarRental)
{
    return this.http.put(`${this.url}/updateCarRental?id=${id}`,carRentalData);
}
deleteCarRental(id:any)
{
    return this.http.delete(`${this.url}/deleteCarRental?id=${id}`);
}
}

File: api/Service/CarRental/CarRentalService.cs
using api.DTO.Car_Rental;
using api.Helpers;
using api.Interfaces.Car_Rentals;

namespace api.Service.CarRental
{
    public class CarRentalService:ICarRentalService
    {
        private readonly ICarRentalRepository _repo;

        public CarRentalService(ICarRentalRepository repo)
        {
            _repo = repo;
        }

        public Task<List<api.Models.Car_Rental.CarRentalEntity>> GetCarRentalsByQuery(CarRentalQueryObjectDto queryObject) =>
            _repo.GetCarRentalsByQuery(queryObject);

        public Task<(List<api.Models.Car_Rental.CarRentalEntity> carRentals, int totalCount)> GetAllCarRentals(int pageNumber, int pageSize) =>
            _repo.GetAllCarRentals(pageSize, pageNumber);

        public Task<List<string>> GetLocations() => _repo.GetLocations();

        public Task<api.Models.Car_Rental.CarRentalEntity> CreateCarRental(CarRentalDto carRentalDTO) => _repo.CreateCarRental(carRentalDTO);

        public Task<api.Models.Car_Rental.CarRentalEntity> UpdateCarRental(int id, CarRentalDto carRentalDTO) => _repo.UpdateCarRental(id, carRentalDTO);

        public Task DeleteCarRental(int id) => _repo.DeleteCarRental(id);

        public Task<List<api.Models.Car_Rental.CarRentalEntity>> SearchByCarName(string name) => _repo.SearchByCarName(name);

        public Task<api.Models.Car_Rental.CarRentalEntity> GetById(int id) => _repo.GetById(id);
    }
}
